### PR TITLE
[sqlite] Add `close` and `deleteAsync` method

### PR DESCRIPTION
**Pull request by Kudo**
# Why

requested from AWS Amplify to add Expo integration for the `@aws-amplify/datastore` package.

# How

- exposing `close` method publicly
- add `deleteAsync` method

# Test Plan

test-suite SQLite
  - added a `should delete db on filesystem from the `deleteAsync()` call` test case

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 1]
```diff
@@ -138,6 +138,16 @@ export declare class SQLError {
 // @docsMissing
 export interface WebSQLDatabase extends Database {
   exec(queries: Query[], readOnly: boolean, callback: SQLiteCallback): void;
+
+  /**
+   * Close the database.
+   */
+  close(): void;
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts:145]:
```suggestion
  closeAsync(): void;
```

File: packages/expo-sqlite/src/SQLite.ts [hunk 2]
```diff
@@ -44,6 +44,14 @@ class SQLiteDatabase {
     this._closed = true;
     ExponentSQLite.close(this._name);
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.ts:45]:
```suggestion
    return ExponentSQLite.close(this._name);
```

File: apps/test-suite/tests/SQLite.js [hunk 3]
```diff
@@ -93,7 +93,7 @@ export function test(t) {
               resolve
             );
           });
-          db._db.close();
+          db.close();
```
Comment (bbarthec) [inline apps/test-suite/tests/SQLite.js:96]:
```suggestion
          db.closeAsync();
```

File: apps/test-suite/tests/SQLite.js [hunk 4]
```diff
@@ -440,5 +441,18 @@ export function test(t) {
...
       });
     }
+
+    if (Platform.OS !== 'web') {
+      t.it('should delete db on filesystem from the `deleteAsync()` call', async () => {
+        const db = SQLite.openDatabase('test.db');
+        let fileInfo = await FS.getInfoAsync(`${FS.documentDirectory}SQLite/test.db`);
+        t.expect(fileInfo.exists).toBeTruthy();
+
+        db.close();
```
Comment (bbarthec) [inline apps/test-suite/tests/SQLite.js:451]:
```suggestion
        await db.closeAsync();
```

**Review by bbarthec — CHANGES_REQUESTED** [review]
Looking good 👍 
My remarks:
- let's name all newly exposed methods following our convention with `Async` sufix
- let's make `close` method return actual promise and let's expose it as `closeAsync`
- let's add some more info to the error messages, like the name of the database that failed to be deleted

**Comment by Kudo** [comment]
> Looking good 👍 My remarks:
> 
> * let's name all newly exposed methods following our convention with `Async` sufix
> * let's make `close` method return actual promise and let's expose it as `closeAsync`
> * let's add some more info to the error messages, like the name of the database that failed to be deleted

thanks for the super great review comment 🤩

File: packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt [hunk 5]
```diff
@@ -58,6 +58,24 @@ class SQLiteModule(private val mContext: Context) : ExportedModule(mContext) {
     promise.resolve(null)
   }
 
+  @ExpoMethod
+  fun deleteAsync(dbName: String, promise: Promise) {
+    val errorCode = "SQLiteError"
+    if (DATABASES.containsKey(dbName)) {
+      promise.reject(errorCode, "Unable to delete opened database '$dbName'")
```
Comment (bbarthec) [inline packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt:65]:
```suggestion
      promise.reject(errorCode, "Unable to delete database '$dbName' that is currently open. Close it prior to deletion.")
```

File: packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt [hunk 6]
```diff
@@ -58,6 +58,24 @@ class SQLiteModule(private val mContext: Context) : ExportedModule(mContext) {
...
+    }
+    val dbFile = File(pathForDatabaseName(dbName))
+    if (!dbFile.exists()) {
+      promise.reject(errorCode, "Database not found")
```
Comment (bbarthec) [inline packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt:69]:
```suggestion
      promise.reject(errorCode, "Database '$dbName' not found")
```

File: packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt [hunk 7]
```diff
@@ -58,6 +58,24 @@ class SQLiteModule(private val mContext: Context) : ExportedModule(mContext) {
...
+      return
+    }
+    if (!dbFile.delete()) {
+      promise.reject(errorCode, "Unable to delete the database file")
```
Comment (bbarthec) [inline packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt:73]:
```suggestion
      promise.reject(errorCode, "Unable to delete the database file for '$dbName' database")
```

File: packages/expo-sqlite/ios/EXSQLite/EXSQLite.m [hunk 8]
```diff
@@ -100,6 +100,37 @@ - (NSValue *)openDatabase:(NSString *)dbName
...
+EX_EXPORT_METHOD_AS(deleteAsync,
+                    deleteDbName:(NSString *)dbName
+                    resolver:(EXPromiseResolveBlock)resolve
+                    rejecter:(EXPromiseRejectBlock)reject)
+{
+  NSString *errorCode = @"E_SQLITE_DELETE_DATABASE";
+
+  @synchronized(self) {
+    if ([cachedDatabases objectForKey:dbName]) {
+      reject(errorCode, [NSString stringWithFormat:@"Unable to delete opened database '%@'", dbName], nil);
```
Comment (bbarthec) [inline packages/expo-sqlite/ios/EXSQLite/EXSQLite.m:112]:
```suggestion
      reject(errorCode, [NSString stringWithFormat:@"Unable to delete database '%@' that is currently open. Close is prior to deletion.", dbName], nil);
```

File: packages/expo-sqlite/ios/EXSQLite/EXSQLite.m [hunk 9]
```diff
@@ -100,6 +100,37 @@ - (NSValue *)openDatabase:(NSString *)dbName
...
+    }
+  }
+
+  NSString *path = [self pathForDatabaseName:dbName];
+  if (!path) {
+    reject(errorCode, @"No FileSystem module.", nil);
+    return;
+  }
+  if (![[NSFileManager defaultManager] fileExistsAtPath:path]) {
+    reject(errorCode, @"Database not found", nil);
```
Comment (bbarthec) [inline packages/expo-sqlite/ios/EXSQLite/EXSQLite.m:123]:
```suggestion
    reject(errorCode, [NSString stringWithFormat:@"Database '%@' not found", dbName], nil);
```

File: packages/expo-sqlite/ios/EXSQLite/EXSQLite.m [hunk 10]
```diff
@@ -100,6 +100,37 @@ - (NSValue *)openDatabase:(NSString *)dbName
...
+    return;
+  }
+  NSError *error;
+  if (![[NSFileManager defaultManager] removeItemAtPath:path error:&error]) {
+    reject(errorCode, @"Unable to delete the database file", error);
```
Comment (bbarthec) [inline packages/expo-sqlite/ios/EXSQLite/EXSQLite.m:128]:
```suggestion
    reject(errorCode, [NSString stringWithFormat:@"Unable to delete the database file for '%@' database", dbName], error);
```

File: packages/expo-sqlite/src/SQLite.ts [hunk 11]
```diff
@@ -42,7 +42,15 @@ class SQLiteDatabase {
 
   close() {
     this._closed = true;
-    ExponentSQLite.close(this._name);
+    return ExponentSQLite.close(this._name);
+  }
+
+  deleteAsync(): Promise<void> {
+    if (!this._closed) {
+      throw new Error('Unable to delete an opening database');
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.ts:50]:
```suggestion
      throw new Error(`Unable to delete '${this._name}' database that is currently open. Close it prior to deletion.`);
```

File: packages/expo-sqlite/src/SQLite.types.ts [hunk 12]
```diff
@@ -138,6 +138,16 @@ export declare class SQLError {
...
 export interface WebSQLDatabase extends Database {
   exec(queries: Query[], readOnly: boolean, callback: SQLiteCallback): void;
+
+  /**
+   * Close the database.
+   */
+  closeAsync(): void;
+
+  /**
+   * Delete the database file.
```
Comment (bbarthec) [inline packages/expo-sqlite/src/SQLite.types.ts:148]:
```suggestion
   * Delete the database file.
   * > The database has to be closed prior to deletion.
```

**Review by bbarthec — COMMENTED** [review]
Looking very good 👍 
I've looked into the error messages once more 👀

**Review by bbarthec — APPROVED** [review]
Superb work 👏

**Pull request merged**